src/lib/tab_groups.ts

```diff
@@ -0,0 +1,330 @@
+import { browserBg, activeTabId } from "./webext"
+
+/**
+ * Return the active window's id.
+ *
+ */
+export async function activeWindowId() {
+    return (await browserBg.windows.getCurrent()).id
+}
+
+/**
+ * Return a set of the current window's tab groups (empty if there are none).
+ *
+ */
+export async function tgroups() {
+    const groups = await browserBg.sessions.getWindowValue(await activeWindowId(), "tridactyl-tgroups")
+    return new Set<string>(groups as string[])
+}
+
+/**
+ * Set the current window's tab groups.
+ *
+ * @param groups The set of groups.
+ *
+ */
+export async function setTgroups(groups: Set<string>) {
+    return browserBg.sessions.setWindowValue(await activeWindowId(), "tridactyl-tgroups", [...groups])
+}
+
+/**
+ * Clear the current window's tab groups.
+ *
+ */
+export async function clearTgroups() {
+    return browserBg.sessions.removeWindowValue(await activeWindowId(), "tridactyl-tgroups")
+}
+
+/**
+ * Return the active tab group for the a window or undefined.
+ *
+ * @param id The id of the window. Use the current window if not specified.
+ *
+ */
+export async function windowTgroup(id?: number) {
+    return browserBg.sessions.getWindowValue(id || await activeWindowId(), "tridactyl-active-tgroup")
```

Comment:
```suggestion
    return browserBg.sessions.getWindowValue(id || await activeWindowId(), "tridactyl-active-tgroup") as string
```
This would avoid the need for for multiple casts in other functions (e.g. tgroupActiveLast/tgroupHandleTabCreated).